docs/topics/gradle/gradle-plugin-variants.md

```diff
@@ -0,0 +1,96 @@
+[//]: # (title: Support for Gradle plugin variants)
+
+Gradle 7.0 introduced a new feature for Gradle plugin authors
+— [plugins with variants](https://docs.gradle.org/7.0/userguide/implementing_gradle_plugins.html#plugin-with-variants).
+This feature makes it easier to add support for latest Gradle features while maintaining compatibility with older Gradle versions
+below 7.1. Learn more about [variant selection in Gradle](https://docs.gradle.org/current/userguide/variant_model.html).
+
+With Gradle plugin variants, the Kotlin team can ship different Kotlin Gradle plugin (KGP) variants for different Gradle versions. 
+The goal is to support the base Kotlin compilation in the `main` variant, which corresponds to the oldest supported versions of
+Gradle. Each variant will have implementations for Gradle features from a corresponding release. The latest variant will
+support the latest Gradle feature set. With this approach, it is possible to extend support for older Gradle versions 
+with limited functionality.
+
+Currently, there are only two variants of the Kotlin Gradle plugin:
+
+* `main` for Gradle versions 6.8.3–6.9.2
+* `gradle70` for Gradle 7.0 and higher
+* `gradle71` for Gradle 7.1-7.4
+* `gradle75` for Gradle 7.5
+* `gradle76` for Gradle 7.6 and higher
+
+In future Kotlin releases, more variants will be probably added.
+
+To check which variant your build uses, enable
+the [`--info` log level](https://docs.gradle.org/current/userguide/logging.html#sec:choosing_a_log_level) and find a
+string in the output starting with `Using Kotlin Gradle plugin`, for example, `Using Kotlin Gradle plugin main variant`.
+
+## Troubleshooting
+
+> Here are workarounds for some known issues with variant selection in Gradle:
+> * [ResolutionStrategy in pluginManagement is not working for plugins with multivariants](https://github.com/gradle/gradle/issues/20545)
+> * [Plugin variants are ignored when a plugin is added as the `buildSrc` common dependency](https://github.com/gradle/gradle/issues/20847)
+>
+{type="note"}
+
+### Gradle can't select a KGP variant in a custom configuration
+
+This is an expected situation that Gradle can't select a KGP variant in a custom configuration.
+If you use a custom Gradle configuration:
+
+```kotlin
+configurations.register("customConfiguraton") {
```

Comment:
@Tapchicoma, I'm sorry, could you please provide the Groovy code for this and the following two scripts?🥺

Reply:
here Groovy will be the same

Reply:
here Groovy will be the same